configure.ac

```diff
@@ -620,6 +622,10 @@ if test "$pac_cross_compiling" = "yes" -a "$ac_cv_prog_cc_cross" = "no" ; then
 fi
 PAC_POP_FLAG([CFLAGS])
 
+# also needed by hwloc in embedded mode, must also come early for expansion
+# ordering reasons
+AC_USE_SYSTEM_EXTENSIONS
```

Comment:
We're enabling all platform and posix extensions now? Seems like a big change.

Reply:
Do we have any control over this? I assume that hwloc requires it.

Reply:
Right, `hwloc` requires this.  In practice, this means that the compiler will not catch some errors that it would have in the past.

Reply:
What would it miss? Wouldn't it only miss the fact that you're now trying to detect whether an extension is supported when it shouldn't be enabled?

I think what we have to most careful about is just making sure that we don't start allowing compiler extensions to get into mainline MPCH.

Reply:
Right, that's the only concern.

I can investigate some more to see why `hwloc` needs it and see if we can maybe convince them to not use those extensions.

Reply:
I assume it's so they can do a bunch of detection to figure out what features they can use or fall back to. Surely they don't require using a certain set of compilers.

Reply:
Yeah, after some more thought this makes perfect sense. The concern remains though, as we've often made an effort to include only platform agnostic code in core MPICH.

Reply:
And by core, I mean the `MPI/MPIR` layers especially.

Reply:
Sorry for all the spam! After close look I see now we can disable this at configure time. I guess the downside is that the extension stuff needs to be done early in configure and not within a conditional...

Reply:
One more comment. Could the way we configure MPL be used to avoid this? https://github.com/pmodels/mpich/blob/master/configure.ac#L1189

Reply:
The way we are building `mpl` is not correct.  Consider the case where `mpl` depends on a different library `libfoo`.  Let's say we are doing static libraries and libtool is not able to embed `libfoo` into `libmpi`.  Now, how can `mpl` tell `mpich` that `-lfoo` needs to be added to `mpicc`?  I don't think we rely on system libraries that way in `mpl`, which is why it is working fine.  But the way `hwloc` allows for embedded builds is the right way to do this anyway.